enhancements/security/openshift-image-policy.md

```diff
@@ -0,0 +1,359 @@
+This enhancement describes how those primatives will be used by OpenShift clusters to secure core workloads:
+
+* The `quay.io/openshift-release-dev/ocp-release` scope, used for OpenShift release images, will require [Sigstore][] signature signed by Red Hat, for any namespace in the cluster.
+* Users may add additional policies if they want, to add additional requirements as they see fit, including for the `quay.io/openshift-release-dev` scope.
+
+## Motivation
+
+As an OpenShift cluster administrator, I want to trust that the release image I'm running comes from Red Hat using the emerging industry standard for doing so: [Sigstore][].
+
+### User Stories
+
+* As a cluster administrator, I want to trust that the OpenShift release image I'm running comes from Red Hat, checking whenever a pod is launched from such an image, and not just at update-time.
+* As a cluster administrator, I want to be able to declare my own additional image policies, including restricting the use of the `quay.io/openshift-release-dev` scope.
+* As a Red Hat CI job maintainer or quality engineer, I have a way to disable Sigstore verification while testing releases before they have been signed.
+
+### Goals
+
+* Define an initial OpenShift ClusterImagePolicy that will be installed in all 4.17 clusters to restrict `quay.io/openshift-release-dev/ocp-release` to Red-Hat-signed images.
+
+### Non-Goals
+
+There are a number of possible next-steps that I'm punting on for now, as I try to nibble off the smallest possible slice of progress.
+This enhancement will not:
+
+* Figure out how to configure a ClusterImagePolicy for verifying a mirrored release image, although this will need to happen before we can use Sigstore signatures in disconnected/restricted-network environments.
+* Suggest changes to mirroring tools, although they will need to learn to mirror Sigstore signatures for use in disconnected/restricted-network environments.
+* Suggest changes to the ImageStream controller, although it will need to learn to mirror Sigstore signatures for verification when running pods from ImageStream sources, e.g. [must-gather][].
+* Suggest policies for `quay.io/openshift-release-dev/ocp-v4.0-art-dev`, although those would help tighten down security for other `openshift-*` namespaces.
+* Support testing unsigned releases as HyperShift hosted clusters on a management cluster that requires signed releases.
+
+## Proposal
+
+This enhancement proposes a new `openshift` ClusterImagePolicy that requires Red Hat signatures for `quay.io/openshift-release-dev/ocp-release` images.
+The new ClusterImagePolicy will be managed by the ClusterVersion operator and applied to all clusters by default (initially tech-preview, gated by the `ImagePolicy` feature gate ([currently `TechPreviewNoUpgrade`][feature-gate]).
```

Comment:
Does that mean we ship the policy per default if image policies are enabled? Or do we want to have an additional toggle switch?

Reply:
Yeah, that's the proposal.  If you think there's a case for a separate feature gate, can you give some pros and cons, for an *Alternatives* section?  Then folks weighing pros and cons could decide to flip that option in as the core proposal and shift "use the `ImagePolicy` feature-gate" out into the *Alternatives* section if they want.  Personally, my initial take is that is makes sense to at least start expecting one gate, and possibly to pivot to a separate gate if the *Implementation order* plan hits trouble during the step (3) testing step, before we've done anything to the payload except (1)'s lifting the current scope restrictions.

Reply:
One more thing to clarify in this section is how we will continue handling gpg verification. In early discussions we had discussed having both side by side initially. We need to decide if we will keep it while this is in tech preview and then disable gpg entirely when we GA sigstore support.

Reply:
Currently this enhancement doesn't suggest changes to the OpenPGP/GPG handling, because:

* We'll need to keep it for the GA 4.17, which won't have the `TechPreviewNoUpgrade` Sigstore stuff.
* It's pretty orthogonal to the Sigstore proposal, so the CVO continuing to do it's GPG thing while the `TechPreviewNoUpgrade` Sigstore stuff happens too is just belt-and-suspenders.  And also helps demo how the transitional period will work if this thing goes GA.

Do you think it's worth calling that out explicitly in Markdown?

Reply:
I do  - a viable plan to migrate, securely and transparently for the end user, over time is an important consideration to the overall proposal.

Reply:
I've pushed 4a5ca6bb -> 7137261e, adding an _Overlap with OpenPGP signing_ section explaining how the two systems are orthogonal and working through some details of a transitional update to make that more clear.

I'm leaving this thread unresolved,  because @saschagrunert 's initial ask about whether this deserves a separate feature gate than the existing `ImagePolicy` feature-gate is not yet clear.  Do folks have any pros or cons to feed an *Alternatives* section about that?  Or can we mark this thread resolved based on reasoning like:

> No real pros/cons yet, but we if we bump into any later like "wow, testing shows this is going to take a long time to get working", we can come split off to a new feature-gate then.